elements/lisk-transactions/src/legacy_cast_votes.ts

```diff
@@ -0,0 +1,121 @@
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { getAddressFromPassphrase } from '@liskhq/lisk-cryptography';
+import {
+	isValidFee,
+	isValidNonce,
+	validateNetworkIdentifier,
+	validatePublicKeys,
+} from '@liskhq/lisk-validator';
+
+import { VoteTransaction } from './11_vote_transaction';
+import { TransactionJSON } from './transaction_types';
+import {
+	createBaseTransaction,
+	prependMinusToPublicKeys,
+	prependPlusToPublicKeys,
+} from './utils';
+
+export interface CastVoteInputs {
+	readonly networkIdentifier: string;
+	readonly nonce: string;
+	readonly fee: string;
+	readonly passphrase?: string;
+	readonly secondPassphrase?: string;
+	readonly unvotes?: ReadonlyArray<string>;
+	readonly votes?: ReadonlyArray<string>;
+}
+
+interface VotesObject {
+	readonly unvotes?: ReadonlyArray<string>;
+	readonly votes?: ReadonlyArray<string>;
+	readonly networkIdentifier: string;
+	readonly fee: string;
+	readonly nonce: string;
+}
+
+const validateInputs = ({
+	fee,
+	nonce,
+	votes = [],
+	unvotes = [],
+	networkIdentifier,
+}: VotesObject): void => {
+	if (!isValidNonce(nonce)) {
+		throw new Error('Nonce must be a valid number in string format.');
+	}
+
+	if (!isValidFee(fee)) {
+		throw new Error('Fee must be a valid number in string format.');
+	}
+
+	if (!Array.isArray(votes)) {
+		throw new Error(
+			'Please provide a valid votes value. Expected an array if present.',
+		);
+	}
+	if (!Array.isArray(unvotes)) {
+		throw new Error(
+			'Please provide a valid unvotes value. Expected an array if present.',
+		);
+	}
+
+	validatePublicKeys([...votes, ...unvotes]);
+
+	validateNetworkIdentifier(networkIdentifier);
+};
+
+export const castVotes = (inputs: CastVoteInputs): Partial<TransactionJSON> => {
+	validateInputs(inputs);
+	const { networkIdentifier, passphrase, votes = [], unvotes = [] } = inputs;
+
+	const plusPrependedVotes = prependPlusToPublicKeys(votes);
+	const minusPrependedUnvotes = prependMinusToPublicKeys(unvotes);
+	const allVotes: ReadonlyArray<string> = [
+		...plusPrependedVotes,
+		...minusPrependedUnvotes,
+	];
+
+	const transaction = {
+		...createBaseTransaction(inputs),
+		type: 11,
+		asset: {
+			// TODO: Remove this after hardfork change. Amount is kept as asset property for exceptions
+			amount: '0',
+			votes: allVotes,
+		},
+	};
+
+	if (!passphrase) {
+		return transaction;
+	}
+
+	const recipientId = getAddressFromPassphrase(passphrase);
+	const transactionWithSenderInfo = {
+		...transaction,
+		// SenderId and SenderPublicKey are expected to be exist from base transaction
```

Comment:
`SenderId and SenderPublicKey expected to exist in the base transaction`

Reply:
same as above